cmd/watch-termination/main.go

```diff
@@ -86,31 +88,49 @@ func run() int {
 		}
 
 		if st, err := os.Stat(*terminationLock); err == nil {
-			klog.Warningf("Previous pod did not terminate gracefully: %s", st.ModTime().String())
+			podName := "unknown"
+			if v := os.Getenv("POD_NAME"); len(v) > 0 {
+				podName = v // pod name is always the same for static pods
+			}
+			msg := fmt.Sprintf("Previous pod %s started at %s did not terminate gracefully", podName, st.ModTime().String())
+
+			klog.Warning(msg)
+			_, _ = terminationLogger.WriteToTerminationLog([]byte(msg + "\n"))
+
 			if client != nil {
 				go wait.PollUntil(5*time.Second, func() (bool, error) {
-					if err := eventf(client.CoreV1().Events(ref.Namespace), *ref, corev1.EventTypeWarning, "NonGracefulTermination", "Previous pod did not terminate gracefully: %s", st.ModTime().String()); err != nil {
+					if err := eventf(client.CoreV1().Events(ref.Namespace), *ref, corev1.EventTypeWarning, "NonGracefulTermination", msg); err != nil {
 						return false, nil
 					}
 
 					select {
 					case <-termCh:
 					default:
-						klog.Infof("Deleting old termination lock file %q", *terminationLock)
-						os.Remove(*terminationLock)
```

Comment:
these lines were executed when the event was successfully created. But the touch command few lines below would conflict with this. We have to delete first, then touch.